### PR TITLE
PullRequest for Issue1705: Make AMDSClientAppController singleton

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
Make AMDSClientAppController singleton so that the individual detectors only need to know about the specific server it wants to talk with.

https://github.com/acquaman/acquaman/issues/1705
Related Acquaman PR:
https://github.com/acquaman/acquaman/pull/1710

### CODE REVIEW

**Comment by XuetaoLiu-S** [comment]
Forgot that I changed the AMDSClietAppController singleton already in https://github.com/acquaman/AcquamanDataServer/pull/81

**Comment by XuetaoLiu-S** [comment]
Need to rework on this...

**Comment by dretrex** [comment]
I know you said you want to rework something about this, but as a first pass things seem alright.
Time Estimate: 5 minutes

File: source/appController/AMDSClientAppController.h [hunk 1]
```diff
@@ -48,24 +49,26 @@ class AMDSClientAppController : public AMDSAppController
...
 	/// request to establish a connection to a specific hostName and the portNumber
+	void connectToServer(const AMDSServerConfiguration &serverConfiguration);
+	/// request to establish a connection to a specific hostName and the portNumber
 	void connectToServer(const QString &hostName, quint16 portNumber);
 	/// request disconnection with a given server identifier
 	void disconnectWithServer(const QString &serverIdentifier);
 
 	/// request data from server for Statistics
-	void requestClientData(const QString &hostName, quint16 portNumber);
+	bool requestClientData(const QString &hostName, quint16 portNumber);
```
Comment (dretrex) [inline source/appController/AMDSClientAppController.h:56]: Only question I have is if these `requestClientData()` methods should be const or not?

**Pull request merged**